compiler/riscv/codegen/RVSystemLinkage.cpp

```diff
@@ -82,134 +54,105 @@ addDependency(
    dep->addPostCondition(vreg, rnum);
    }
 
+TR::RVSystemLinkageProperties::RVSystemLinkageProperties()
+   {
+   _properties = IntegersInRegisters|FloatsInRegisters|RightToLeft;
+
+   _registerFlags[TR::RealRegister::zero] = Preserved|RV_Reserved; // zero
+   _registerFlags[TR::RealRegister::ra]   = Preserved|RV_Reserved; // return address
+   _registerFlags[TR::RealRegister::sp]   = Preserved|RV_Reserved; // sp
+   _registerFlags[TR::RealRegister::gp]   = Preserved|RV_Reserved; // gp
+   _registerFlags[TR::RealRegister::tp]   = Preserved|RV_Reserved; // tp
+
+   _registerFlags[TR::RealRegister::t0]   = Preserved|RV_Reserved; // fp
+   _registerFlags[TR::RealRegister::t1]   = 0;
+   _registerFlags[TR::RealRegister::t2]   = 0;
+
+   _registerFlags[TR::RealRegister::s0]   = Preserved;
+   _registerFlags[TR::RealRegister::s1]   = Preserved;
```

Comment:
This is because registers are defined in architectural order. In RISC-V backend I decided to use "ABI" names instead of architectural names (`x0` .. `x31`). 

In RISC-V, `t2` is `x7`, `s0` is `x8`, `s1` is `x9`, `a0` is `x10` and so on and `s2` is `x18` and so on. This is I believe to allow use of `s0` and `s1` in compressed instruction set (something that we may want to implement at some point to improve code density, should not be so hard). 

I'll add a comment stating that flags are defined in architectural order and point to ABI spec.